### ISSUE TITLE
UpdateUserAttributeResult.isUpdated is true on Android but false on iOS if the attribute requires confirmation

### ISSUE THREAD

**Issue by Jordan-Nelson** [aws-amplify/amplify-flutter#609]
**Describe the bug**
The response from `Amplify.Auth.updateUserAttribute()` is a `UpdateUserAttributeResult`. The result has a boolean isUpdated property. This value is true on Android in all scenarios. On iOS it is false if the attribute requires confirmation.

**To Reproduce**
1. Run the following code on iOS and observe that the printed value is false.
```dart
var res = await Amplify.Auth.updateUserAttribute(
  userAttributeKey: 'email',
  value: 'email@email.com',
);
print('is updated: ${res.isUpdated}');
```
2. Run the same code on Android and observe that the printed value is true

**Expected behavior**
Consistency between the two platforms.

**Screenshots**
These screenshots were taken while debugging in xcode and Android studio, and show the different values coming back from amplify-ios and amplify-android

Android
![Screen Shot 2021-06-02 at 11 58 07 AM](https://user-images.githubusercontent.com/20613561/120514826-b2ad9680-c39b-11eb-9bbe-4e3fe0feb8c3.png)

iOS
<img width="868" alt="Screen Shot 2021-06-02 at 12 04 02 PM" src="https://user-images.githubusercontent.com/20613561/120514829-b2ad9680-c39b-11eb-9af5-bbb613458360.png">

**Platform**
Amplify Flutter current supports iOS and Android.  This issue is reproducible in (check all that apply):
[x] Android
[x] iOS

**Additional context**
amplify-ios and amplify-android are returning different values (see screenshots above). Related issue: https://github.com/aws-amplify/amplify-flutter/issues/581

**Comment by Jordan-Nelson** [comment]
For anyone facing this issue, the current workaround is to ignore `isUpdated`, and use `nextStep` instead. This value will be consistent across platforms.

**Comment by Jordan-Nelson** [comment]
For anyone facing this issue, we have recently released a new developer preview version of the Auth category ([amplify_flutter v1.0.0-next.0](https://pub.dev/packages/amplify_flutter/versions/1.0.0-next.0)). This new version is a full re-write of the auth category in dart, and it supports Mobile, Desktop, and Web. This issue is addressed in this new version.

You can read more about the developer preview release in the [blog post](https://aws.amazon.com/blogs/mobile/amplify-flutter-auth-web-and-desktop/) and [docs](https://docs.amplify.aws/lib/auth/getting-started/q/platform/flutter/). I will update this issue again once this new version is out of developer preview.

**Comment by Jordan-Nelson** [comment]
This issue has been addressed in v1.0.0 of Amplify Flutter, which is now stable. This release also includes web and desktop support for Auth, API, Analytics, and Storage. You can see the list of new features and bug fixes in the [release notes](https://github.com/aws-amplify/amplify-flutter/releases/tag/v1.0.0), and see more details on how to migrate in the [upgrade guide](https://docs.amplify.aws/lib/project-setup/upgrade-guide/q/platform/flutter/).
